test/puppetlabs/trapperkeeper/services/authorization/authorization_core_test.clj

```diff
@@ -69,6 +69,25 @@
    :match-request {:path "/path" :type "path"}})
 
 (deftest invalid-configs-fail
+  (testing "Configuration section required"
+    (is (thrown-with-msg?
+         IllegalArgumentException
+         #"Missing authorization service configuration."
+         (validate-auth-config! nil))))
+
+  (testing "Version number required in configuration"
+    (is (thrown-with-msg?
+         IllegalArgumentException
+         #"Unsupported or missing version in configuration file.*"
+         (validate-auth-config! {:rules []}))))
+
+  (testing "Rule names must be unique"
+    (is (thrown-with-msg?
+         IllegalArgumentException
+         #".* Rules must be uniquely named."
+         (validate-auth-config! {:version 1
+                                 :rules [testrule testrule]}))))
```

Comment:
I started to add a shared helper function in `testutils` that contained the top-level configuration map which could be shared by the service and core tests, but the result wasn't worth it.  If we add more tests and more defaults we could make the refactoring.